### ISSUE TITLE
[1.46] UnicodeDecodeError on Windows 7 

### ISSUE THREAD

**Issue by PLNech** [nvbn/thefuck#254]
See the stacktrace:

```
[~] cd Work/friends/simon                                                                   
cd:cd:13: no such file or directory: Work/friends/simon
[~] fuck  
Traceback (most recent call last):
  File "C:\python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\thefuck.exe\__main__.py", line 9, in <module>
  File "C:\python27\lib\site-packages\thefuck\main.py", line 129, in main
    command = get_command(settings, sys.argv)
  File "C:\python27\lib\site-packages\thefuck\main.py", line 86, in get_command
    result.stderr.read().decode('utf-8'))
  File "C:\python27\lib\encodings\utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x8a in position 15: invalid start byte
```

**Comment by WalkMe** [comment]
+1

	 $ fuck
	 Traceback (most recent call last):
	  File "C:\Python27\lib\runpy.py", line 162, in _run_module_as_main
		"__main__", fname, loader, pkg_name)
	  File "C:\Python27\lib\runpy.py", line 72, in _run_code
		exec code in run_globals
	  File "c:\Python27\Scripts\thefuck.exe\__main__.py", line 9, in <module>
	  File "C:\Python27\lib\site-packages\thefuck\main.py", line 77, in main
		fix_command()
	  File "C:\Python27\lib\site-packages\thefuck\main.py", line 28, in fix_command
		selected_command = select_command(corrected_commands)
	  File "C:\Python27\lib\site-packages\thefuck\ui.py", line 106, in select_command
		logs.confirm_text(selector.value)
	  File "C:\Python27\lib\site-packages\thefuck\logs.py", line 61, in confirm_text
		blue=color(colorama.Fore.BLUE)))
	  File "C:\Python27\lib\encodings\cp862.py", line 12, in encode
		return codecs.charmap_encode(input,errors,encoding_map)
	UnicodeEncodeError: 'charmap' codec can't encode character u'\u2191' in position
	 45: character maps to <undefined>

**Comment by legendtang** [comment]
I've also encounter this issue. The first time running this command will freeze forever.

P.S. Using latest Git for Windows 2.8.3

````
$ fuck
Traceback (most recent call last):
  File "d:\python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "d:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "D:\Python27\Scripts\thefuck.exe\__main__.py", line 9, in <module>
  File "d:\python27\lib\site-packages\thefuck\main.py", line 80, in main
    fix_command()
  File "d:\python27\lib\site-packages\thefuck\main.py", line 26, in fix_command
    command = types.Command.from_raw_script(sys.argv[1:])
  File "d:\python27\lib\site-packages\thefuck\types.py", line 120, in from_raw_script
    stderr = result.stderr.read().decode('utf-8')
  File "d:\python27\lib\encodings\utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xb2 in position 9: invalid start byte

````

**Comment by scorphus** [comment]
Hi, I'll try and reproduce this error as soon as I get the chance. Thanks
for reporting it.
Em 31/05/2016 00:20, "Wensheng Tang" <notifications@github.com> escreveu:

**Comment by aendinalt** [comment]
The same issue
```
Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "c:\Python27\Scripts\thefuck.exe\__main__.py", line 9, in <module>
  File "c:\python27\lib\site-packages\thefuck\main.py", line 82, in main
    fix_command()
  File "c:\python27\lib\site-packages\thefuck\main.py", line 26, in fix_command
    command = types.Command.from_raw_script(sys.argv[1:])
  File "c:\python27\lib\site-packages\thefuck\types.py", line 120, in from_raw_s
cript
    stderr = result.stderr.read().decode('utf-8')
  File "c:\python27\lib\encodings\utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xad in position 6: invalid s
tart byte
```

**Comment by nitriques** [comment]
Yup same issue for me

$ fuck
Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\thefuck.exe\__main__.py", line 9, in <module>
  File "c:\python27\lib\site-packages\thefuck\main.py", line 51, in main
    fix_command(known_args)
  File "c:\python27\lib\site-packages\thefuck\main.py", line 27, in fix_command
    command = types.Command.from_raw_script(raw_command)
  File "c:\python27\lib\site-packages\thefuck\types.py", line 125, in from_raw_script
    stderr = result.stderr.read().decode('utf-8')
  File "c:\python27\lib\encodings\utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x82 in position 84: invalid start byte

**Comment by Tomotoes** [comment]
The same issue
![image](https://user-images.githubusercontent.com/32236122/51546205-9e3d6300-1e9e-11e9-8a4a-fa9c495824e8.png)

**Comment by gracefullight** [comment]
same on windows10

```
Traceback (most recent call last):
  File "d:\scoop\apps\python\3.7.2\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "d:\scoop\apps\python\3.7.2\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "D:\Scoop\apps\python\current\scripts\thefuck.exe\__main__.py", line 9, in <module>
  File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\entrypoints\main.py", line 26, in main
    fix_command(known_args)
  File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\entrypoints\fix_command.py", line 36, in fix_command
    command = types.Command.from_raw_script(raw_command)
  File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\types.py", line 82, in from_raw_script
    output = get_output(script, expanded)
  File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\output_readers\__init__.py", line 20, in get_output
    return rerun.get_output(script, expanded)
  File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\output_readers\rerun.py", line 62, in get_output
    output = result.stdout.read().decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc0 in position 4: invalid start byte
```

**Comment by moziar** [comment]
> same on windows10
> 
> ```
> Traceback (most recent call last):
>   File "d:\scoop\apps\python\3.7.2\lib\runpy.py", line 193, in _run_module_as_main
>     "__main__", mod_spec)
>   File "d:\scoop\apps\python\3.7.2\lib\runpy.py", line 85, in _run_code
>     exec(code, run_globals)
>   File "D:\Scoop\apps\python\current\scripts\thefuck.exe\__main__.py", line 9, in <module>
>   File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\entrypoints\main.py", line 26, in main
>     fix_command(known_args)
>   File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\entrypoints\fix_command.py", line 36, in fix_command
>     command = types.Command.from_raw_script(raw_command)
>   File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\types.py", line 82, in from_raw_script
>     output = get_output(script, expanded)
>   File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\output_readers\__init__.py", line 20, in get_output
>     return rerun.get_output(script, expanded)
>   File "d:\scoop\apps\python\3.7.2\lib\site-packages\thefuck\output_readers\rerun.py", line 62, in get_output
>     output = result.stdout.read().decode('utf-8')
> UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc0 in position 4: invalid start byte
> ```

Any idea to slove it? I've got the same error

**Comment by scorphus** [comment]
Fixed in c2df71caed917ea659a72a393b105881ec0709df
